rest/src/main/java/com/redhat/ipaas/rest/v1/dao/DataManager.java

```diff
@@ -257,7 +266,7 @@ public boolean delete(String kind, String id) {
             throw new EntityNotFoundException("Can not find " + kind + " with id " + id);
 
         WithId entity = cache.get(id);
-        if (entity != null && doWithDataAccessObject(kind, d -> d.delete(entity))) {
+        if (doWithDataAccessObject(kind, d -> d.delete(entity)) || entity != null ) {
```

Comment:
Don't think the DAO throws an exception in that case (it should return false).  But your comment did highlight the fact that doWithDataAccessObject can return null, so I need to handle that case.